modules/ztp-image-based-upgrade-installing-lifecycle-agent-using-gitops.adoc

```diff
@@ -0,0 +1,332 @@
+├── source-crs
+│   ├── custom-crs
+│   │   ├── OadpSubscriptionNS.yaml
+│   │   ├── OadpSubscriptionOperGroup.yaml
+│   │   ├── OadpSubscription.yaml
+----
+--
+
+. Add the CRs to your common `PolicyGenTemplate`.
++
+.Example directory structure
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "example-common-latest"
+  namespace: "ztp-common"
+spec:
+  bindingRules:
+    common: "true"
+    du-profile: "latest"
+  sourceFiles:
+    - fileName: custom-crs/OadpSubscriptionNS.yaml
+      policyName: "subscriptions-policy"
+    - fileName: custom-crs/OadpSubscriptionOperGroup.yaml
+      policyName: "subscriptions-policy"
+    - fileName: custom-crs/OadpSubscription.yaml
+      policyName: "subscriptions-policy"
+[...]
+----
+
+. Create the `DataProtectionApplication` CR and the S3 secret.
+
+.. Create the following CRs in your `source-crs/custom-crs` directory:
++
+--
+.DataProtectionApplication.yaml
+[source,yaml]
+----
+apiVersion: oadp.openshift.io/v1
+kind: DataProtectionApplication
+metadata:
+  name: dataprotectionapplication
+  namespace: openshift-adp
+  annotations:
+    ran.openshift.io/ztp-deploy-wave: "100"
+spec:
+  configuration:
+    restic:
+      enable: false <1>
+    velero:
+      defaultPlugins:
+        - aws
+        - openshift
+      resourceTimeout: 10m
+  backupLocations:
+    - velero:
+        config:
+          profile: "default"
+          region: minio
+          s3Url: $url
+          insecureSkipTLSVerify: "true"
+          s3ForcePathStyle: "true"
+        provider: aws
+        default: true
+        credential:
+          key: cloud
+          name: cloud-credentials
+        objectStorage:
+          bucket: $bucketName
+          prefix: $prefixName
+status:
+  conditions:
+  - reason: Complete
+    status: "True"
+    type: Reconciled
+----
+<1> The `spec.configuration.restic.enable` field must be set to `false` for an image-based upgrade because persistent volume contents are retained and reused after the upgrade.
+
+.OadpSecret.yaml
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cloud-credentials
+  namespace: openshift-adp
+  annotations:
+    ran.openshift.io/ztp-deploy-wave: "100"
+type: Opaque
+----
+
+.OadpBackupStorageLocationStatus.yaml
+[source,yaml]
+----
+apiVersion: velero.io/v1
+kind: BackupStorageLocation
+metadata:
+  namespace: openshift-adp
+  annotations:
+    ran.openshift.io/ztp-deploy-wave: "100"
+status:
+  phase: Available
+----
+
+The `OadpBackupStorageLocationStatus.yaml` CR verifies the availability of backup storage locations created by OADP.
+--
+
+.. Add the CRs to your site `PolicyGenTemplate` with overrides.
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "example-cnf"
+  namespace: "ztp-site"
+spec:
+  bindingRules:
+    sites: "example-cnf"
+    du-profile: "latest"
+  mcp: "master"
+  sourceFiles:
+    ...
+    - fileName: OadpSecret.yaml
```

Comment:
I think the original commit was fine (no need for "custom-crs") since we should have this in ZTP? (same for the other "custom-crs/" changes...we can revert them)

Please checkout @Missxiaoguo latest PR and just copy the examples from there https://github.com/openshift-kni/cnf-features-deploy/pull/1945 ?

Reply:
These source crs have now pushed to 4.16 ztp container, but we are good for now. We will need to update this a bit postGA when ready

Reply:
Ah good to know! (ignore my previous comment..."custom-crs/" is good in that case!)